### PR TITLE
[bugfix 1866718] fix no icon for new add connector unless close and reopen intellij.

### PR DESCRIPTION
**Pull request by shenqianjin**
[AB#1866718](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1866718)

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/azure/toolkit/intellij/connector/database/DatabaseResourcePanel.java [hunk 1]
```diff
@@ -96,6 +97,9 @@ private void init() {
         testResultTextPane.setEditable(false);
         testConnectionButton.setEnabled(false);
         envPrefixTextField.setText(definition instanceof SqlServerDatabaseResource.Definition ? "AZURE_SQL_" : "AZURE_MYSQL_");
+        this.passwordSaveComboBox.setValue(Arrays.stream(Password.SaveType.values())
+                .filter(e -> StringUtils.equals(e.name(), AzureConfigurations.getInstance().passwordSaveType())).findAny()
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/azure/toolkit/intellij/connector/database/DatabaseResourcePanel.java]: use az.config instead

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/azure/toolkit/intellij/connector/database/DatabaseResourcePanel.java [hunk 2]
```diff
@@ -226,9 +230,9 @@ public DatabaseResource getData() {
         password.password(inputPasswordField.getPassword());
         password.saveType(passwordSaveComboBox.getValue());
 
-        final DatabaseResource resource = new DatabaseResource(definition.getType(),
-                definition instanceof SqlServerDatabaseResource.Definition ?
-                ((SqlDatabaseEntity) databaseComboBox.getValue()).getId() : ((MySqlDatabaseEntity) databaseComboBox.getValue()).getId());
+        final DatabaseResource resource = definition instanceof SqlServerDatabaseResource.Definition ?
+                new SqlServerDatabaseResource(((SqlDatabaseEntity) databaseComboBox.getValue()).getId()) :
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/azure/toolkit/intellij/connector/database/DatabaseResourcePanel.java:234]: can we force databaseComboBox getValue returns IAzureResourceEntity?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/azure/toolkit/intellij/connector/database/DatabaseResourcePanel.java (hunk 1 above)
Comment (shenqianjin) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/azure/toolkit/intellij/connector/database/DatabaseResourcePanel.java]: updated.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/azure/toolkit/intellij/connector/database/DatabaseResourcePanel.java (hunk 2 above)
Comment (shenqianjin) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/azure/toolkit/intellij/connector/database/DatabaseResourcePanel.java:234]: no. will extract a IDatabaseResourceEntity to refactor this codes in the Database rewfactor task/PR very soon.

**Review by andxu — APPROVED** [review]
LGTM

**Pull request merged**